packages/protocol/contracts/common/FeeCurrencyWhitelist.sol

```diff
@@ -46,7 +53,51 @@ contract FeeCurrencyWhitelist is
    * @return Patch version of the contract.
    */
   function getVersionNumber() external pure returns (uint256, uint256, uint256, uint256) {
-    return (1, 1, 1, 0);
+    return (1, 1, 2, 0);
+  }
+
+  // removed with address zero
+  function setAdaptor(address token, address adaptor) external onlyOwner {
+    adapters[token] = adaptor;
+    emit AdaptorSet(token, adaptor);
+  }
+
+  // TODO make external
+  function getAdaptor(address underlyingToken) public view returns(address){
+    address result = adapters[underlyingToken];
+    return (result != address(0))? result: underlyingToken;
+  }
```

Comment:
in this case the if i call getAdaptor with an random address the random address will be returned.  

from what we said last night we wanted to restrict to valid ones right?

Reply:
It's actually hard to check now if something is or not a whitelisted gas fee address as it's an array and not a mapping. You'll get that but it doesn't mean anything.

Reply:
fair enough. 

In that case Id argue it should return address(0) when nothing is found. The method is getAdapter() so it returning null when either address is wrong or when address does not need work is intuitive.